src/mongo/mongoConnectionStrings.ts

```diff
@@ -63,23 +63,16 @@ export async function parseMongoConnectionString(connectionString: string): Prom
         }
     }
 
-    const serverConfig: Server | ReplSet | Mongos = mongoClient.db(testDb).serverConfig;
-    // Azure CosmosDB comes back as a ReplSet
-    if (serverConfig instanceof ReplSet) {
-        // get the first connection string from the servers for the ReplSet
-        // this may not be best solution, but the connection (below) gives
-        // the replicaset host name, which is different than what is in the connection string
-        // "s" is not part of ReplSet static definition but can't find any official documentation on it. Yet it is definitely there at runtime. Grandfathering in.
-        // tslint:disable-next-line:no-any
-        const rs: any = serverConfig;
-        host = rs.s.options.servers[0].host;
-        port = rs.s.options.servers[0].port;
-    } else {
-        // tslint:disable-next-line: no-any
-        host = (<any>serverConfig).host;
-        // tslint:disable-next-line: no-any
-        port = (<any>serverConfig).port;
-    }
+    const serverConfig = mongoClient.db(testDb).serverConfig;
+
+    // get the first connection string from the servers list
+    // this may not be best solution, but the connection (below) gives
+    // host name of single server, mongos instance or the primany from replicaSet which is different than what is in the connection string (espcially for Replica sets)
+    // "s" is not part of the static definition but can't find any official documentation on it. Yet it is definitely there at runtime. Grandfathering in.
+    // tslint:disable-next-line:no-any
+    const rs: any = serverConfig;
+    host = rs.s.options.servers[0].host;
```

Comment:
`s.options.servers ` property is available on all types of accounts. I have tested with local mongo connection (single server) and the referred code works for single server type as well.